src/components/Button/Button.scss

```diff
@@ -2,25 +2,36 @@
 @import "~office-ui-fabric/src/components/Button/Button";
 
 .ms-Button--icon {
+  @include focus-border(0px);
   background-color: transparent;
+  color: $ms-color-neutralSecondary;
   padding: 0;
   min-width: auto;
   height: auto;
   border: 0;
 
-  &:hover {
+  &:hover,
+  &:active {
     background-color: transparent;
     border-color: transparent;
     outline-color: transparent;
+    color: $ms-color-neutralPrimary;
   }
```

Comment:
Double checked with design to make sure this is the desired look. Also, worked with design to figure out high contrast colors for icon buttons. Decided best to use colors that match up with link colors in high contrast. Per design no hover color change necessary in high contrast. Cursor changing to pointer is change enough.